### PR TITLE
wasm: update proxy-wasm-cpp-host to use refactored code around runtimes.

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

ref: https://github.com/proxy-wasm/proxy-wasm-cpp-host/pull/152

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -948,10 +948,10 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "WebAssembly for Proxies (C++ host implementation)",
         project_desc = "WebAssembly for Proxies (C++ host implementation)",
         project_url = "https://github.com/proxy-wasm/proxy-wasm-cpp-host",
-        version = "aba2704bcd7d8adce2ccaf07c4ecbaf0cffeb7ef",
-        sha256 = "6263b45f87fc0c106eb88f31238553b1bde68f8a5ea25e20e95a5e6624a896d2",
+        version = "d9511116f4d68c5c201a3705d92cc79c5a483f05",
+        sha256 = "79847f9784b73a2a8aee1a6a70ca405f46ff7d05fd26fc918295e82ef8d9a392",
         strip_prefix = "proxy-wasm-cpp-host-{version}",
-        urls = ["https://github.com/proxy-wasm/proxy-wasm-cpp-host/archive/{version}.tar.gz"],
+        urls = ["https://github.com/mathetake/proxy-wasm-cpp-host/archive/{version}.tar.gz"],
```
Comment (htuch) [inline bazel/repository_locations.bzl:954]: Why this change? We should be using the official proxy-wasm repo.

Comment (mathetake) [reply bazel/repository_locations.bzl:954]: This is intentionally set to my fork to run test against the ongoing PR in cpp-host and that’s why I marked this WIP PR for now 🙏

Comment (PiotrSikora) [reply bazel/repository_locations.bzl:954]: FYI: on GitHub, you can refer to a commit from a fork even in the "main" repo, so you don't need to change the URL.

Comment (mathetake) [reply bazel/repository_locations.bzl:954]: i didn't know that.. thanks..!
